modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankEvalSpec.java

```diff
@@ -120,9 +152,51 @@ public void setSpecifications(Collection<RatedRequest> specifications) {
         } , REQUESTS_FIELD);
     }
 
+    public static RankEvalSpec parse(XContentParser parser, RankEvalContext context, boolean templated) throws IOException {
+        RankEvalSpec spec = PARSER.parse(parser, context);
+
+        if (templated) {
+            logger.trace("template: {}",  spec.script);
+            for (RatedRequest query_spec : spec.getSpecifications()) {
+                @SuppressWarnings({ "unchecked", "rawtypes" })
+                Map<String, String> params = (Map) query_spec.getParams();
```

Comment:
We can - but then we cannot reuse

https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java#L413

and

https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/common/xcontent/XContentParser.java#L130

without casting. Or am I missing something?